workflow/controller/operator.go

```diff
@@ -1037,6 +1037,36 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 			node.Phase = wfv1.NodeError
 		} else {
 			node.Outputs = &outputs
+			tmplCtx, err := woc.createTemplateContext(node.GetTemplateScope())
+			if err != nil {
+				log.Errorf("Failed to create template context for node %s", node.ID)
```

Comment:
please log error using `log.WithError(err)`

Reply:
should you actually return this error?

Reply:
the only issue I have with returning this error is `assessNodeStatus` doesn't return an error atm, is this a change worth making to the fxn as a whole?